src/Tools/dotnet-monitor/CollectionRules/Configuration/CustomShortcutsConfigureNamedOptions.cs

```diff
@@ -0,0 +1,214 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Actions;
+using Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Options;
+using Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Triggers;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+
+namespace Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Configuration
+{
+    internal sealed class CustomShortcutsConfigureOptions :
+        IPostConfigureOptions<CollectionRuleOptions>
+    {
+        private readonly ICollectionRuleTriggerOperations _triggerOperations;
+        private readonly ICollectionRuleActionOperations _actionOperations;
+        private readonly IOptionsMonitor<CustomShortcutOptions> _shortcutOptions;
+        private readonly IConfiguration _configuration;
+        private readonly ILogger<CustomShortcutsConfigureOptions> _logger;
+
+        public CustomShortcutsConfigureOptions(
+            ICollectionRuleTriggerOperations triggerOperations,
+            ICollectionRuleActionOperations actionOperations,
+            IOptionsMonitor<CustomShortcutOptions> shortcutOptions,
+            IConfiguration configuration,
+            ILogger<CustomShortcutsConfigureOptions> logger)
+        {
+            _triggerOperations = triggerOperations;
+            _actionOperations = actionOperations;
+            _shortcutOptions = shortcutOptions;
+            _configuration = configuration;
+            _logger = logger;
+        }
+
+        public void PostConfigure(string name, CollectionRuleOptions options)
```

Comment:
I don't think the concerns are separated correctly. I would expect that there is an `IPostConfigureOptions<CustomShortcutOptions>` implementation that effectively does the same thing as https://github.com/dotnet/dotnet-monitor/blob/main/src/Tools/dotnet-monitor/CollectionRules/Configuration/CollectionRuleConfigureNamedOptions.cs where it binds the settings for each action and trigger. Then this `IPostConfigureOptions<CollectionRuleOptions>` can just assign the filters, trigger, actions, limits wholesale without have to bind anything itself.

Reply:
So, I just pushed changes that I _think_ hopefully reflects what you're getting at; I didn't do a lot of testing, but it isn't a huge change and it passes the tests (and a small amount of manual testing). Let me know if I misinterpreted what you had in mind.